src/routes/v1/productRoutes.js

```diff
@@ -1,14 +1,18 @@
 const express = require("express");
-const uploadImage = require("../../middleware/upload");
 
 const router = express.Router();
 const productController = require("../../controllers/productController");
 const catchAsync = require("../../core/catchAsync");
 const { isAdminOrSeller, isProductAuthor } = require("../../middleware/auth");
+const uploadImage = require("../../middleware/upload");
 
 router.get("/", catchAsync(productController.getAllProducts));
 
-router.get("/new", isAdminOrSeller, productController.showNewForm);
+router.get("/form", isAdminOrSeller, productController.showAddProductForm);
+
+router.get("/:id/form", isAdminOrSeller, productController.showEditProductForm);
```

Comment:
Why `/:id/form` ? Why REST convention is not followed here?

Reply:
for to get a HTML form filled with product details, I'm using the endpoint GET /product/:id/edit